demo.ipynb

```diff
@@ -10,6 +10,94 @@
fb/TJtHwIECBAgQKAyBASNlVHnzTVLQWMiL2hMMDQJECBAgAABAgQIECBAgACBshMQNJZdSYtqQoLGpByCxgRDkwABAgQIECBAgAABAgQIECg7AUFj2ZW0qCYkaEzKIWhMMDQJECBAgAABAgQIECBAgACBshMQNJZdSYtqQoLGpByCxgRDkwABAgQIECBAgAABAgQIECg7AUFj2ZW0qCYkaEzKIWhMMDQJECBAgAABAgQIECBAgACBshMQNJZdSYtqQoLGpByCxgRDkwABAgQIECBAgAABAgQIECg7AUFj2ZW0qCYkaEzKIWhMMDQJECBAgAABAgQIECBAgACBshMQNJZdSYtqQoLGpBytQeOvfvWrOPLII5M9mgQIECBAgAABAgQIECBAgACB0hfIBo1Tp06NlpaWtsn85je/ieOPP77ttwaBTRUQNCZyrUFjskmTAAECBAgQIECAAAECBAgQIFDWAoLGsi5vQScnaEy4Z8yYEVdddVU0Nja2bZ0zZ0489NBDbb81CBAgQIAAAQIECBAgQIAAAQLlJCBoLKdqbt65CBo78b/11lvjrLPO6uQouwkQIECAAAECBAgQIECAAAECpSkgaCzNuhXjqAWNnVRF0NgJkN0ECBAgQIAAAQIECBAgQIBASQsIGku6fEU1eEFjJ+UQNHYCZDcBAgQIECBAgAABAgQIECBQ0gKCxpIuX1ENXtDYSTmy72tcu3ZtJ0fZTYAAAQIECBAgQIAAAQIECBAoTYEBAwZEdXV1aQ7eqItKQNBYVOUwGAIECBAgQIAAAQIECBAgQIAAAQKlKSBoLM26GTUBAgQIECBAgAABAgQIECBAgACBohIQNBZVOQyGAAECBAgQIECAAAECBAgQIECAQGkKCBpLs25GTYAAAQIECBAgQIAAAQIECBAgQKCoBASNRVUOgyFAgAABAgQIECBAgAABAgQIECBQmgKCxtKsm1ETIECAAAECBAgQIECAAAECBAgQKCoBQWNRlcNgCBAgQIAAAQIECBAgQIAAAQIECJSmgKCxNOtm1AQIECBAgAABAgQIECBAgAABAgSKSkDQWFTlMBgCBAgQIECAAAECBAgQIECAAAECpSkgaCzNuhk1AQIECBAgQIAAAQIECBAgQIAAgaISEDQWVTkMhgABAgQIECBAgAABAgQIECBAgEBpCggaS7NuRk2AAAECBAgQIECAAAECBAgQIECgqAQEjUVVDoMhQIAAAQIECBAgQIAAAQIECBAgUJoCgsbSrJtREyBAgAABAgQIECBAgAABAgQIECgqAUFjUZXDYAgQIECAAAECBAgQIECAAAECBAiUpoCgsTTrZtQECBAgQIAAAQIECBAgQIAAAQIEikpA0FhU5TAYAgQIECBAgAABAgQIECBAgAABAqUpIGgszboZNQECBAgQIECAAAECBAgQIECAAIGiEhA0FlU5DIYAAQIECBAgQIAAAQIECBAgQIBAaQoIGkuzbkZNgAABAgQIECBAgAABAgQIECBAoKgEBI1FVQ6DIUCAAAECBAgQIECAAAECBAgQIFCaAoLG0qybURMgQIAAAQIECBAgQIAAAQIECBAoKgFBY1GVw2AIECBAgAABAgQIECBAgAABAgQIlKaAoLE062bUBAgQIECAAAECBAgQIECAAAECBIpKQNBYVOUwGAIECBAgQIAAAQIECBAgQIAAAQKlKSBoLM26GTUBAgQIECBAgAABAgQIECBAgACBohIQNBZVOQyGAAECBAgQIECAAAECBAgQIECAQGkKCBpLs25GTYAAAQIECBAgQIAAAQIECBAgQKCoBASNRVUOgyFAgAABAgQIECBAgAABAgQIECBQmgKCxtKsm1ETIECAAAECBAgQIECAAAECBAgQKCoBQWNRlcNgCBAgQIAAAQIECBAgQIAAAQIECJSmgKCxNOtm1AQIECBAgAABAgQIECBAgAABAgSKSkDQWFTlMBgCBAgQIECAAAECBAgQIECAAAECpSkgaCzNuhk1AQIECBAgQIAAAQIECBAgQIAAgaISEDQWVTkMhgABAgQIECBAgAABAgQIECBAgEBpCggaS7NuRk2AAAECBAgQIECAAAECBAgQIECgqAQEjUVVDoMhQIAAAQIECBAgQIAAAQIECBAgUJoCgsbSrJtREyBAgAABAgQIECBAgAABAgQIECgqAUFjUZXDYAgQIECAAAECBAgQIECAAAECBAiUpoCgsTTrZtQECBAgQIAAAQIECBAgQIAAAQIEikpA0FhU5TAYAgQIECBAgAABAgQIECBAgAABAqUpIGgszboZNQECBAgQIECAAAECBAgQIECAAIGiEhA0FlU5DIYAAQIECBAgQIAAAQIECBAgQIBAaQoIGkuzbkZNgAABAgQIECBAgAABAgQIECBAoKgEBI1FVQ6DIUCAAAECBAgQIECAAAECBAgQIFCaAoLG0qybURMgQIAAAQIECBAgQIAAAQIECBAoKgFBY1GVw2AIECBAgAABAgQIECBAgAABAgQIlKaAoLE062bUBAgQIECAAAECBAgQIECAAAECBIpKQNBYVOUwGAIECBAgQIAAAQIECBAgQIAAAQKlKSBoLM26GTUBAgQIECBAgAABAgQIECBAgACBohIQNBZVOQyGAAECBAgQIECAAAECBAgQIECAQGkKCBpLs25GTYAAAQIECBAgQIAAAQIECBAgQKCoBASNRVUOgyFAgAABAgQIECBAgAABAgQIECBQmgKCxtKsm1ETIECAAAECBAgQIECAAAECBAgQKCoBQWNRlcNgCBAgQIAAAQIECBAgQIAAAQIECJSmgKCxNOtm1AQIECBAgAABAgQIECBAgAABAgSKSkDQWFTlMBgCBAgQIECAAAECBAgQIECAAAECpSkgaCzNuhk1AQIECBAgQIAAAQIECBAgQIAAgaISEDQWVTkMhgABAgQIECBAgAABAgQIECBAgEBpCvx/CJZwifXuHHkAAAAASUVORK5CYII=)"
```

Comment:
I expected this image to have "C_i", not "C_i^{[m]}", since the covariates do not depend on m.

Reply:
In general the covariates can depend on m (see my email).

Reply:
Ah, I see. The implementation in the code assumes the covariates for a given node are constant, but I guess there's nothing conceptually wrong with them varying across samples. I've added a comment to that effect in https://github.com/geraschenko/gmrfs/pull/2/commits/1bb3e9a4a279458bef63b5b0e3e1bb658f6831fb to avoid people like me getting confused.